apis/Google.Storage.V1/Google.Storage.V1/UpdateBucketOptions.cs

```diff
@@ -51,20 +52,38 @@ public sealed class UpdateBucketOptions
         /// </summary>
         public PredefinedObjectAcl? PredefinedDefaultObjectAcl { get; set; }
 
-        internal void ModifyRequest(UpdateRequest request)
+        /// <summary>
+        /// If set to true, no other preconditions must be set, and
+        /// the local metageneration of the bucket being updated is not used
+        /// to create a precondition.
+        /// </summary>
+        public bool? ForceNoPreconditions { get; set; }
+
+        internal void ModifyRequest(UpdateRequest request, Bucket bucket)
         {
             if (IfMetagenerationMatch != null && IfMetagenerationNotMatch != null)
             {
                 throw new ArgumentException($"Cannot specify {nameof(IfMetagenerationMatch)} and {nameof(IfMetagenerationNotMatch)} in the same options", "options");
             }
+            if (ForceNoPreconditions == true && (IfMetagenerationMatch != null || IfMetagenerationNotMatch != null))
+            {
+                throw new ArgumentException($"Cannot specify {nameof(ForceNoPreconditions)} and any explicit precondition in the same options", "options");
+            }
 
-            if (IfMetagenerationMatch != null)
+            if (ForceNoPreconditions != true && IfMetagenerationMatch == null && IfMetagenerationNotMatch == null)
             {
-                request.IfMetagenerationMatch = IfMetagenerationMatch;
+                request.IfMetagenerationNotMatch = bucket.Metageneration;
```

Comment:
Yes, that was a brain-dead typo. Hopefully would have caught it when writing tests :)